### PR TITLE
Jetpack Section: Remove Jetpack Feature Flags

### LINKED ISSUE #13267: Jetpack Section: Project Issue
This is the main project issue for the Jetpack Section project

### Tasks
- [x] Create feature flags for Activity Log, Restore, Backup and Scan
- [x] Activity Log Filters: https://github.com/wordpress-mobile/WordPress-Android/issues/13268
- [x] Restore: #13328 
- [x] Backup Download: #13329 
- [x] Scan: https://github.com/wordpress-mobile/WordPress-Android/issues/13326
- [x] Backup Screen: #13629

 #### Activity Log
- [x] Final Design Review 
- [x] Release
- [x] Remove feature flags two weeks after the release

#### Restore
- [x] Final Design Review
- [x] Release
- [x] Remove feature flags two weeks after the release

 #### Backup
- [x] Final Design Review 
- [x] Release
- [x] Remove feature flags two weeks after the release

 #### Scan
- [x] Final Design Review 
- [x] Release
- [x] Remove feature flags two weeks after the release

### PR DESCRIPTION
**Pull request by ParaskP7**
Fixes #13267

This PR is responsible for removing all `Jetpack Section` related feature flags from the codebase. After that, the whole `Jetpack Section` grandparent related issue (see fixes) can be closed since those are the last items on its todo list.

To test:
- Launch the app and go to `My Site` tab.
- Test the `Activity Log` and `Backup` screens and more specifically make sure the `Activity Log Filters` are available and functional.
- Test the `Backup` screen individually and more specifically make sure the screen is available and functional.
- Test the `Scan` screen and more specifically make sure the screen is available and functional.
- Test the `Back Download` screen and flows from within the `Activity Log` and `Backup` screens.
- Test the `Restore` screen and flows from within the `Activity Log` and `Backup` screens.

Merge instructions:
- Wait for confirmation from @leandroalonso before merging.
- Wait for a 👍 from @malinajirka as well.

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

N/A

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ParaskP7** [comment]
👋 @leandroalonso !

I want to bring this PR to your attention so as you can give the 🟢 light that we can now get rid of all the `Jetpack Section` related feature flags from our codebase. It has been some time since we release and it seems that now it is a good time to remove those. Wdyt?

PS: 👋 @malinajirka ! It would be great if we got your 👍 as well! 🙏

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java [hunk 1]
```diff
@@ -369,14 +369,4 @@ public static boolean hasCustomDomain(@NonNull SiteModel site) {
     public static boolean hasFullAccessToContent(@Nullable SiteModel site) {
         return site != null && (site.isSelfHostedAdmin() || site.getHasCapabilityEditPages());
     }
-
-    // TODO: Inline this method when legacy MySiteFragment is removed
-    public static boolean isScanEnabled(boolean scanFeatureFlagEnabled, boolean scanPurchased, SiteModel site) {
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:374]: ❓ I'm wondering if we should wait with removal of this method until MySiteFragment gets replaced with the ImprovedMySiteFragment. Someone might not realize both of these fragment exist and update the condition in just one of them. Wdyt?

**Review by malinajirka — COMMENTED** [review]
Thanks for working on this @ParaskP7! I've scanned through the changes and they look good to me. Having said that I didn't do a full review, let me know if you'd prefer me to do a full review.

**Review by zwarm — APPROVED** [review]
@ParaskP7 - All looks good. I tested the the Jetpack features including on the `my_site_improvements_enabled` flag. Thanks for the cleanup too!

Feel free to merge once you get the 👍 .

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java (hunk 1 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:374]: Good point @malinajirka. If you decide to change this @ParaskP7 , I can run through my tests again. Just let me know.

Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:374]: 👋 Jirka!

Yeap, I was thinking about that too, I think it is better for me to revert that change, we already talked about that when that got introduced in the codebase. I just wanted to clean this up as well, but, it is better to be on the safe side for now.

PS: @zwarm , I'll do this tomorrow and let you know. 🙏

**Comment by leandroalonso** [comment]
@ParaskP7 🟢 :shipit:

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java (hunk 1 above)
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:374]: 👋 @malinajirka @zwarm !

This is now done here: b7d33855cfb97d0fa3cf67b1c4708ab421ed80a0

Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java:374]: Thanks for making the change and it works as expected. 👍

**Pull request merged**